movement/watch_faces/complication/probability_face.c

```diff
@@ -0,0 +1,181 @@
+const uint16_t DICE_TYPES[] = {2, 4, 6, 8, 10, 12, 20, 100};
+
+
+// --------------
+// Custom methods
+// --------------
+
+static void display_dice_roll(probability_state_t *state) {
+    char buf[7]; // Num chars needed + 1 for \0
+    if (state->rolled_value == 0) {
+        if (state->dice_sides == 100) {
+            sprintf(buf, " C    ", state->dice_sides);
+        } else {
+            sprintf(buf, "%2d    ", state->dice_sides);
+        }
+    } else if (state->dice_sides == 2) {
+        if (state->rolled_value == 1) {
+            sprintf(buf, "%2d   H", state->dice_sides);
+        } else {
+            sprintf(buf, "%2d   T", state->dice_sides);
+        }
+    } else if (state->dice_sides == 100) {
+        sprintf(buf, " C %3d", state->rolled_value);
+    } else {
+        sprintf(buf, "%2d %3d", state->dice_sides, state->rolled_value);
+    }
+    watch_display_string(buf, 4);
+}
+
+// According to SO this is the better way to roll a random number to avoid introducing skew.
+// https://stackoverflow.com/a/2999130
+// (As opposed to using modulus like: `state->rolled_value = rand() % state->dice_sides + 1;`)
+static void generate_random_number(probability_state_t *state) {
+    int limit = state->dice_sides - 1;
+    int divisor = RAND_MAX/(limit+1);
+    int retval;
+
+    do {
+        retval = rand() / divisor;
+    } while (retval > limit);
+
+    // Add 1 to shift from 0-based to actual roll number
+    state->rolled_value = retval + 1;
+}
+
+static void display_dice_roll_animation(probability_state_t *state) {
+    if (state->is_rolling) {
+        if (state->animation_frame == 0) {
+            watch_display_string("   ", 7);
+            watch_set_pixel(1, 4);
+            watch_set_pixel(1, 6);
+            state->animation_frame = 1;
+        } else if (state->animation_frame == 1) {
+            watch_clear_pixel(1, 4);
+            watch_clear_pixel(1, 6);
+            watch_set_pixel(2, 4);
+            watch_set_pixel(0, 6);
+            state->animation_frame = 2;
+        } else if (state->animation_frame == 2) {
+            watch_clear_pixel(2, 4);
+            watch_clear_pixel(0, 6);
+            watch_set_pixel(2, 5);
+            watch_set_pixel(0, 5);
+            state->animation_frame = 3;
+        } else if (state->animation_frame == 3) {
+            state->animation_frame = 0;
+            state->is_rolling = false;
+            movement_request_tick_frequency(1);
+            display_dice_roll(state);
+        }
+    }
+}
+
+
+// ---------------------------
+// Standard watch face methods
+// ---------------------------
+void probability_face_setup(movement_settings_t *settings, uint8_t watch_face_index, void ** context_ptr) {
+    (void) settings;
+    if (*context_ptr == NULL) {
+        *context_ptr = malloc(sizeof(probability_state_t));
+        memset(*context_ptr, 0, sizeof(probability_state_t));
+
+        // Seed random number generator
+        srand(time(NULL));
```

Comment:
You can remove this line entirely if you switch to using arc4random, since it'll regularly pull entropy from the true hardware RNG.

Reply:
Just an FYI on this, it seems since I was doing `#include <time.h>` in this file it would result in this error when I tried to make a uf2 build:
```LD build/watch.elf
/opt/gcc-arm-none-eabi/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: /opt/gcc-arm-none-eabi/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/lib/thumb/v6-m/nofp/libc.a(lib_a-gettimeofdayr.o): in function `_gettimeofday_r':
gettimeofdayr.c:(.text._gettimeofday_r+0xe): undefined reference to `_gettimeofday'
collect2: error: ld returned 1 exit status
make: *** [../../rules.mk:24: build/watch.elf] Error 1```

Hoping that removing that include and using arc4random instead should give me the result I want.